torch/cuda/__init__.py

```diff
@@ -60,7 +60,37 @@
         if not _version.hip:
             import pynvml  # type: ignore[import]
         else:
-            import amdsmi  # type: ignore[import]
+            import ctypes
+
+            _original_CDLL = ctypes.CDLL  # type: ignore[misc,assignment]
```

Comment:
There is already a `torch._load_global_defs()` function that deals with preloading various dependencies for CUDA
https://github.com/pytorch/pytorch/blob/4c5a9a5f949ef3019fc3ef095034ccfc973ff13d/torch/__init__.py#L303

Why not `try: CDLL.dlopen("libami_smi.so");catch:` there and just leave a note here that this is done in order to avoid the situation? Because libraries are cached by their filename, so if dynamic linker finds there is already one instance of `libfoo.so` loaded it will return reference to that instance next time dlopen is called

Reply:
Spoke offline, but inlining here too for history: `libamd_smi.so` is referred to by absolute path in `amdsmi`, so preloading won't work